### PR TITLE
TEST: clear a few warnings in model builder tests

### PR DESCRIPTION
**Pull request by david-cortes-intel**
### Description

This PR clears a few warnings that are observed when executing unit tests that involve building and converting models from XGBoost and LightGBM, and removes an unnecessary `print` statement that executes in one of the tests.

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.  
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

File: tests/test_model_builders.py [hunk 1]
```diff
@@ -229,6 +231,31 @@ def setUpClass(cls):
 
 @unittest.skipUnless(shap_supported, reason=shap_not_supported_str)
 class XGBoostClassificationModelBuilder(unittest.TestCase):
+    def getConversionWarningIsExpected(self):
```
Comment (Alexsandruss) [inline tests/test_model_builders.py:234]: Use snake case.

File: tests/test_model_builders.py [hunk 2]
```diff
@@ -229,6 +231,31 @@ def setUpClass(cls):
...
+        if (
+            self.xgb_model._estimator_type == "classifier"
+            and self.xgb_model.objective == "binary:logitraw"
+        ):
+            return True
+        else:
+            return False
+
+    @contextlib.contextmanager
+    def conditionalAssertWarns(self, warning):
```
Comment (Alexsandruss) [inline tests/test_model_builders.py:244]: Same here.

File: tests/test_model_builders.py [hunk 3]
```diff
@@ -229,6 +231,31 @@ def setUpClass(cls):
...
+        else:
+            with warnings.catch_warnings() as ctx:
+                warnings.simplefilter("error")
+                try:
+                    yield [ctx]
+                finally:
+                    pass
+
     @classmethod
     def setUpClass(cls, base_score=0.5, n_classes=2, objective="binary:logistic"):
```
Comment (Alexsandruss) [inline tests/test_model_builders.py:260]: Same here.

**Review by Alexsandruss — CHANGES_REQUESTED** [review]

File: tests/test_model_builders.py (hunk 1 above)
Comment (david-cortes-intel) [reply tests/test_model_builders.py:234]: Changed to snake case.

**Comment by Alexsandruss** [comment]
@razdoburdin, pending on your review.

File: tests/test_model_builders.py [hunk 4]
```diff
@@ -456,9 +495,12 @@ def setUpClass(cls):
             "learning_rage": 0.05,
             "metric": {"l2", "l1"},
             "verbose": -1,
-            "n_estimators": 1,
         }
-        cls.lgbm_model = lgbm.train(params, train_set=lgbm.Dataset(X_train, y_train))
+        cls.lgbm_model = lgbm.train(
+            params,
+            train_set=lgbm.Dataset(X_train, y_train),
+            num_boost_round=1,
```
Comment (razdoburdin) [inline tests/test_model_builders.py:502]: why do you prefer setting `num_boost_round` to `n_estimators`?

**Review by razdoburdin — APPROVED** [review]
looks ok
